setup.py

```diff
@@ -39,14 +39,12 @@
 
 
 import datetime
-from distutils import log
-from distutils.command.build import build
-from distutils.dep_util import newer
-from distutils.spawn import find_executable
 import glob
 from io import StringIO
+import logging as log
```

Comment:
Is this import needed?

Reply:
Yes, log is used at various places in setup.py in the different commands. `distutils` provided its own `log` module, the recommended replacement is using the `logging` module instead, see https://peps.python.org/pep-0632/#migration-advice .